flink-python/pyflink/fn_execution/beam/beam_coder_impl_slow.py

```diff
@@ -144,15 +144,17 @@ def __repr__(self):
 
 class RowCoderImpl(FlattenRowCoderImpl):
 
-    def __init__(self, field_coders):
+    def __init__(self, field_coders, field_names):
         super(RowCoderImpl, self).__init__(field_coders)
+        self.field_names = field_names
 
     def encode_to_stream(self, value: Row, out_stream, nested):
         self._encode_one_row_to_stream(value, out_stream, nested)
 
     def decode_from_stream(self, in_stream, nested):
         row_kind_value, fields = self._decode_one_row_from_stream(in_stream, nested)
         row = Row(*fields)
+        row._fields = self.field_names
```

Comment:
What about add a method set_field_names to Row?

Reply:
Yes.Make sensse